website/src/app/page.tsx

```diff
@@ -0,0 +1,271 @@
+    }, 3000);
+
+    return () => clearInterval(interval);
+  });
+
+  return (
+    <div className="m-12 items-center justify-center font-mono lg:text-2xl text-md lg:flex">
+      <p className="flex w-full justify-center p-4">
+        Verify <span className="font-bold">&nbsp;{signatureType}&nbsp;</span>
+        signatures on-chain.
+      </p>
+    </div>
+  );
+}
+
+function ButtonsBar({
+  toggleDetailView,
+}: {
+  toggleDetailView: (view: DetailViewOptions) => void;
+}) {
+  return (
+    <div className="justify-center items-center my-12 grid text-center lg:max-w-1/2 lg:grid-cols-3">
+      <button
+        className="mx-6 group rounded-lg border px-5 py-4 transition-colors border-gray-300 hover:bg-gray-100"
+        onClick={() => {
+          toggleDetailView(DetailViewOptions.Deployments);
+        }}
+      >
+        <h2 className={`mb-3 text-2xl font-semibold`}>
+          Deployments{" "}
+          <span className="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none">
+            &darr;
+          </span>
+        </h2>
+        <p className={`m-0 max-w-[30ch] text-sm opacity-50`}>
+          Find on-chain deployments.
+        </p>
+      </button>
+
+      <button
+        className="mx-6 group rounded-lg border px-5 py-4 transition-colors border-gray-300 hover:bg-gray-100"
+        onClick={() => {
+          toggleDetailView(DetailViewOptions.Usage);
+        }}
+      >
+        <h2 className={`mb-3 text-2xl font-semibold`}>
+          Usage{" "}
+          <span className="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none">
+            &darr;
+          </span>
+        </h2>
+        <p className={`m-0 max-w-[30ch] text-sm opacity-50`}>
+          See Solidity usage code.
+        </p>
+      </button>
+
+      <a
+        href="https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
+        className="mx-6 group rounded-lg border px-5 py-4 transition-colors border-gray-300 hover:bg-gray-100"
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        <h2 className={`mb-3 text-2xl font-semibold`}>
+          Blog{" "}
+          <span className="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none">
+            -&gt;
+          </span>
+        </h2>
+        <p className={`m-0 max-w-[30ch] text-sm opacity-50`}>Learn more.</p>
+      </a>
+    </div>
+  );
+}
+
+function UsageTab() {
+  const usageCode = `bytes32 hash; // message hash
+uint256 r, s; // signature
+uint256 x, y; // public key
+
+address verifier = 0xc2b78104907F722DABAc4C69f826a522B2754De4;
+bytes memory args = abi.encode(hash, r, s, x, y);
+(bool success, bytes memory ret) = verifier.staticcall(args);
+assert(success); // never reverts, always returns 0 or 1
+bool valid = abi.decode(ret, (uint256)) == 1;`;
```

Comment:
recommend simplifying to something like

```solidity
import "p256-verifier/Verify.sol";
```
...
```solidity
bytes32 hash; // message hash
uint256 r, s; // signature
uint256 x, y; // public key

bool valid = p256VerifySignature(hash, r, s, x, y);
```

...then that function can include the malleability check

Reply:
unrelated to the website but Verify.sol could expose, say

`p256VerifySignature` and `p256VerifySignatureAllowMalleability`

Reply:
good point; done here: https://github.com/daimo-eth/p256-verifier/pull/21